pkg/querier/vcs/service.go

```diff
@@ -189,13 +193,41 @@ func (q *Service) GetCommit(ctx context.Context, req *connect.Request[vcsv1.GetC
 		return nil, err
 	}
 
-	commit, err := ghClient.GetCommit(ctx, gitURL.GetOwnerName(), gitURL.GetRepoName(), req.Msg.Ref)
+	owner := gitURL.GetOwnerName()
+	repo := gitURL.GetRepoName()
+	ref := req.Msg.GetRef()
+
+	commit, err := q.tryGetCommit(ctx, ghClient, owner, repo, ref)
 	if err != nil {
 		return nil, err
 	}
+
 	return connect.NewResponse(commit), nil
 }
 
+// tryGetCommit attempts to retrieve a commit using different ref formats (commit hash, branch, tag).
+// It tries each format in order and returns the first successful result.
+func (q *Service) tryGetCommit(ctx context.Context, client gitHubCommitGetter, owner, repo, ref string) (*vcsv1.GetCommitResponse, error) {
+	refFormats := []string{
+		ref,            // Try as a commit hash
+		"heads/" + ref, // Try as a branch
+		"tags/" + ref,  // Try as a tag
+	}
```

Comment:
Eventually I think we will want to try be smarter about using the SHA, heads, or tags ref. Someone could theoretically have the same name for a branch and tag.

I don't think we should address this here, as this PR is enough of an improvement already, but it is something to think of for the future.